### DISCUSSION TITLE
Extracting Boundary Nodes in 2D Mesh

### DISCUSSION THREAD

**Discussion by wandastr** [libMesh/libmesh#3326] [Q&A]
How do I extract only the boundary points of a 2D Mesh? I would like to iterate through the boundary points so that I compute the distance from another object to the mesh. I successfully created the mesh by calling triangle within LibMesh.

**Comment by roystgnr** [comment]
You can use `BoundaryInfo::sync()` and then iterate over its nodes, but then you'll be creating boundary edge elements you don't need in the process.  You might find it easiest to just loop over all elements yourself, test for null `neighbor_ptr(s)` on each side, and loop over `nodes_on_side(s)` after to add node pointers to your own container.
